text/0000-hash/0000-hash.md

```diff
@@ -0,0 +1,302 @@
++ Feature name: `bee-hash`
++ Start date: 2019-10-15
++ RFC PR: [iotaledger/bee-rfcs#0000](https://github.com/iotaledger/bee-rfcs/pull/0000)
++ Bee issue: [iotaledger/bee#0000](https://github.com/iotaledger/bee/issues/0000)
+
+# Summary
+
+<!-- TODO: 81 rounds? rounds of what? -->
+
+This RFC proposes the implementation of the cryptographic hash functions `CurlP` and `Kerl`. `CurlP81`, a newtype for
+`CurlP` with 81 rounds, and `Kerl` are the two cryptographic hash functions used by `iri v1.8.1`, the IOTA reference
+implementation. Both are sponge constructions, and thus follow a very similar architecture.
+
+In `iri v1.8.1`, `CurlP81` is used for generating transaction hashes and --- through the `PearlDiver` algorithm --- for
+proof of work calculations. `Kerl` on the other hand is used as part of all other operations requiring cryptographic
+security, namely address generation, signature generation and verification, and bundle hash calculation.
```

Comment:
If this is supposed to be exhaustive, we should mention that the coordinator also uses Kerl to produce milestones on the Mainnet.